qcodes/parameters/parameter_base.py

```diff
@@ -307,6 +312,55 @@ def __init__(
 
             instrument.parameters[name] = self
 
+    def _build__doc__(self) -> str | None:
+        return self.__doc__
+
+    @property
+    def vals(self) -> Validator | None:
+        if len(self._vals):
+            return self._vals[0]
+        else:
+            return None
+
+    @vals.setter
+    def vals(self, vals: Validator | None) -> None:
+        if vals is not None and len(self._vals) > 0:
+            self._vals[0] = vals
```

Comment:
i think this case still needs a test, right? a non-trivial one))

Reply:
This case is not new but existing functionality that should already be tested but sure why not add a new one

Reply:
well, the exsting tests don't assert for the case where there are more than 2 validators - so the assertion would be that the 2nd validator stays instact after `.vals` is set to something new